pkg/api/presenters/kind.go

```diff
@@ -11,8 +11,12 @@ func ObjectKind(i interface{}) *string {
 	switch i.(type) {
 	case api.Consumer, *api.Consumer:
 		result = "Consumer"
+	case api.ConsumerList, *api.ConsumerList, []api.Consumer, []*api.Consumer:
+		result = "ConsumerList"
```

Comment:
we don't actually have this kind "ConsumerList".  It _is_ an openapi struct for making a list of objects, but our only Kind in the API is Consumer.

Reply:
see our internal api-guidelines (that can probably be ported to github, AFAIK):  https://gitlab.cee.redhat.com/service/api-guidelines#representation

Reply:
Thank you for sharing the link. I comprehend how a single instance is represented using 'Customer.' However, I'm curious about the representation for a collection of instances. I noticed a response example that retrieves a list of accounts at https://gitlab.cee.redhat.com/service/api-guidelines#representation, in this case, the representation for collections of accounts is designated as `AccountList`:

```json
{
  "kind": "AccountList",
  "items": [
    {
      "kind": "Account",
      "id": "123",
      "href": "/api/account_mgmt/v1/accounts/123",
      "name": "myaccount",
      ...
    },
    {
      "kind": "Account",
      "id": "456",
      "href": "/api/account_mgmt/v1/accounts/456",
      "name": "youraccount",
      ...
    },
    ...
  ]
}
```

Reply:
my mistake. i was misreading the Kind on the object _in_ the list. 

but we also [don't have the *List objects listed in the presenters](https://gitlab.cee.redhat.com/service/uhc-account-manager/-/blob/master/pkg/api/presenters/kind.go?ref_type=heads) in AMS and I wondered why.

It seems like AMS hard-codes the *List kind in each handler's List function ([account list, for example](https://gitlab.cee.redhat.com/service/uhc-account-manager/-/blob/master/pkg/handlers/account.go?ref_type=heads#L181)) , since each list is a bespoke composition.

I will resolve my comments. I think what you have here is correct.

Reply:
yes, AMS hard-codes the *List kind in handler functions.